ansible/roles/machine/provision_ipaserver/tasks/main.yml

```diff
@@ -5,6 +5,111 @@
     dest: /usr/local/bin/ipa-run-webui-tests
     mode: 0755
 
+- block:
+    - name: set PKI related variables
+      set_fact:
+        pki_dir: "/tmp/pki"
+        pki_pass: "Secret123"
+        pki_host: "{{ inventory_hostname_short }}"
+        pki_org: "Example Org"
+        meta_fqdn: "{{ inventory_hostname }}.ipa.test"
+        meta_domain: "{{ inventory_hostname }}.ipa.test"
+
+    - name: Install pyOpenSSL
+      dnf:
+        name: python3-pyOpenSSL
+        state: present
+
+    - name: Create PKI directory
+      file:
+        path: "{{ pki_dir }}/ca1"
+        state: directory
+        mode: '0600'
+
+    - name: Generate private keys
+      openssl_privatekey:
+        path: "{{ pki_dir }}/{{ item }}.key"
+        size: 2048
+      with_items:
+        - ca1
+        - ca1/{{ pki_host }}
+
+    - name: Generate server CSRs
+      openssl_csr:
+        path: "{{ pki_dir }}/ca1/{{ item }}.csr"
+        privatekey_path: "{{ pki_dir }}/ca1/{{ item }}.key"
+        common_name: "{{ meta_fqdn }}"
+        subject:
+          commonName: "{{ meta_fqdn }}"
+          organizationName: "{{ pki_org }}"
+      with_items:
+        - "{{ pki_host }}"
+
+    - name: Generate CA CSR
+      openssl_csr:
+        path: "{{ pki_dir }}/ca1.csr"
+        privatekey_path: "{{ pki_dir }}/ca1.key"
+        common_name: CA
+        basic_constraints:
+          - CA:TRUE
+          - pathlen:0
+        basic_constraints_critical: yes
+        key_usage:
+          - cRLSign
+          - digitalSignature
+          - keyCertSign
+          - nonRepudiation
+        key_usage_critical: yes
+        subject:
+          commonName: CA
+          organizationName: "{{ pki_org }}"
```

Comment:
With these settings the CA subject is `CN = CA, CN = CA, O = Example Org`

Reply:
Hello Flo! Thank you for your feedback. I am not quite sure here what's the desired subject and how to achieve it.